nexus/tests/integration_tests/instances.rs

```diff
@@ -5218,6 +5329,85 @@ pub async fn instance_wait_for_state_as(
     }
 }
 
+/// Waits for an instance's VMM to be registered with the simulated sled-agent.
+///
+/// This is necessary when *restarting* an instance, as the instance's external
+/// API state will be `Starting` as soon as a VMM record is created in the
+/// database, even if that VMM's internal state is `VmmState::Creating` (which
+/// means it has not yet been registered with the sled-agent). If we attempt to
+/// simulate an instance before the simulated sled-agent has registered its VMM,
+/// the simulated sled-agent will panic.
```

Comment:
Yeah...I think doing that would complicate the simulated sled-agent pretty
substantially, because when we're in this state, the sled-agent hasn't yet been
_told_ it's supposed to know about that instance, but there's no way for it to
distinguish between not having been told _yet_ and an instance that was never
supposed to exist. But, perhaps it's okay for the simulate API to return
different errors than the "real" instance-get and instance-post APIs and so
on...the current thing still feels like less work, though...